vembrane/modules/tag.py

```diff
@@ -0,0 +1,220 @@
+from sys import stderr
+from typing import Dict, Iterator, Set
+
+from pysam.libcbcf import VariantFile, VariantHeader, VariantRecord
+
+from .. import __version__
+from ..common import (
+    AppendKeyValuePair,
+    AppendTagExpression,
+    check_expression,
+    normalize,
+    single_outer,
+    swap_quotes,
+)
+from ..errors import FilterAlreadyDefined, FilterTagNameInvalid, VembraneError
+from ..representations import Environment
+from .filter import DeprecatedAction, read_auxiliary
+
+
+def add_subcommand(subparsers):
+    parser = subparsers.add_parser("tag")
+    parser.register("action", "deprecated", DeprecatedAction)
+    parser.add_argument(
+        "--tag",
+        "-t",
+        help="Tag records using the FILTER field.",
+        nargs=1,
+        metavar="TAG=EXPRESSION",
+        action=AppendTagExpression,
+        required=True,
+    )
+    parser.add_argument(
+        "vcf", help="The file containing the variants.", nargs="?", default="-"
+    )
+    parser.add_argument(
+        "--output",
+        "-o",
+        default="-",
+        help="Output file, if not specified, output is written to STDOUT.",
+    )
+    parser.add_argument(
+        "--output-fmt",
+        "-O",
+        default="vcf",
+        choices=["vcf", "bcf", "uncompressed-bcf"],
+        help="Output format.",
+    )
+    parser.add_argument(
+        "--annotation-key",
+        "-k",
+        metavar="FIELDNAME",
+        default="ANN",
+        help="The INFO key for the annotation field.",
+    )
+    parser.add_argument(
+        "--aux",
+        "-a",
+        nargs=1,
+        action=AppendKeyValuePair,
+        metavar="NAME=PATH",
+        default={},
+        help="Path to an auxiliary file containing a set of symbols",
+    )
+    parser.add_argument(
+        "--overwrite-number-info",
+        nargs=2,
+        action="append",
+        metavar=("FIELD", "NUMBER"),
+        default=[],
+        help="Overwrite the number specification for INFO fields "
+        "given in the VCF header. "
+        "Example: `--overwrite-number cosmic_CNT .`",
+    )
+    parser.add_argument(
+        "--overwrite-number-format",
+        nargs=2,
+        action="append",
+        metavar=("FIELD", "NUMBER"),
+        default=[],
+        help="Overwrite the number specification for FORMAT fields "
+        "given in the VCF header. "
+        "Example: `--overwrite-number-format DP 2`",
+    )
+    parser.add_argument(
+        "--tag-mode",
+        "-m",
+        default="pass",
+        choices=["pass", "fail"],
+        help="Set, whether to tag records that pass the tag expression(s), "
+        "or records that fail them."
+        "By default, records for which the tag expression(s) pass are tagged. "
+        "This allows for descriptive tag names such as `q_at_least_30` "
+        "to correspond to `QUAL >= 30`. "
+        "However, the VCF specification (`v4.4`) defines tags to be set when a "
+        "filter expression is failed.",
```

Comment:
Slightly improve my wording, maybe?
```suggestion
        help="Set, whether to tag records that pass the tag expression(s), "
        "or records that fail them."
        "By default, vembrane tags records for which the tag expression(s) pass. "
        "This allows for descriptive tag names such as `q_at_least_30`, "
        "which would correspond to an expression `QUAL >= 30`. "
        "However, the VCF specification (`v4.4`) defines tags to be set when a "
        "filter expression is failed, so vembrane also offers the `fail` mode.",
```